core/src/queue.rs

```diff
@@ -49,8 +49,16 @@ pub enum Error {
     #[error("Transaction is already applied")]
     InBlockchain,
     /// Signature condition check failed
-    #[error("Failure during signature condition execution")]
-    SignatureCondition(#[from] Report),
+    #[error("{0}")]
+    SignatureCondition(#[from] SignatureConditionError),
+}
+
+/// Signature condition check error
```

Comment:
Very good. Now we have a lot more info.

Reply:
One thing you can consider (and not necessarily implement) is a struct-like variant. 

We don't use them in IntoSchema structs, because it's not something that other languages support, but in this case, it might save you creating a struct, if all you need is two named fields.

Reply:
Done